### ISSUE TITLE
How about keeping just the core (max,+) algebraic part in the package and moving the systems-related part into a separate package?

### ISSUE THREAD

**Issue by hurak** [Lecrapouille/MaxPlus.jl#9]
I was just wondering: How about keeping just the core (max,+) algebraic part in the package and moving the discrete-event systems-related part into a separate package? 

In contrast to some languages like Matlab, here in Julia we pay no price for having the functionality split into several (and possibly even a larger number of) smaller packages. The package manager in Julia can manage the dependencies easily and – and that is crucial – transparently for the user. 

In case you find the idea reasonable, how about name such system-related package as something like `MaxPlusLinearSystems.jl`? 

But it is not urgent at all. It is just that now as you are refreshing and updating the package, it is perhaps an opportunity for such move. And, in fact, it can make further maintaining and developing easier. Just my 2 cents.

**Comment by Lecrapouille** [comment]
@hurak Do you mean to move syslin.jl and flowshop.jl apart and create their own repo ? Why not ! But the Scilab toolbox did not offer more than "composing" systems (parallel, serie ... explicit and implicit). In my sense would be more logical that a new repo MaxPlusLinearSystems.jl solves/offers more things than what this current toolbox offers.

**Comment by hurak** [comment]
Yes, that is what I meant. I view this max-plus algebra simply as "another algebra", similar to linear algebra, that is, computation with matrices and vectors, for which we have `LinearAlgebra` in Julia. I do not expect anything more from it. Then we have some other package that have `LinearAlgebra` as their dependency. For example, `ControlSystems,jl` package that defines a state-space model of a system and some basic manipulation and computation. 

The fact that initially the package `MaxPlusLinearSystems.jl` will not contain much is not a problem, is it? Smaller packages are fine too. But the mission of the two packages will be quite clear. And perhaps some more goodies will come later. I can think of adding some optimization-based control design (in the spirit of Bart De Schutter's papers). And on the other hand, guys who are not interested in the system-theoretic applications will have a clean package just for max-plus algebra where they can get oriented quickly and start using or even contributing.

**Comment by Lecrapouille** [comment]
@hurak ok ! but let's do this in two times: stabilize this dev branch with all current files and merge it to master first :)
